### ISSUE TITLE
Is the `Notify` name prefix necessary?

### ISSUE THREAD

**Issue by johnhooks** [WordPress/wp-feature-notifications#190]
Since this is a feature plugin, and we are adding directly to the core `/wp-json/wp/v2` rest route, is the prefix `notify` necessary in class, interface, and file names? 

### Revision

Should not initially use the `/wp/v2` api namespace. Variables, functions and classes should have a similar prefix.

**Comment by johnhooks** [comment]
This is related to https://github.com/WordPress/wp-feature-notifications/issues/4 and https://github.com/WordPress/wp-feature-notifications/issues/194 , it would be good to chose one prefix and use through out the project.

**Comment by johnhooks** [comment]
There is currently inconsistencies in the naming of functions, classes and variables in the code base.

There are prefixes of:

- `WP_NOTIFICATION_CENTER` - In `wp-feature-notifications.php`
- `WP_Notify` - In most of the `includes/` directory.
- `WP\Notifications\REST` - Namespace in `includes/rest-api/class-notification-controller.php`

**Comment by erikyo** [comment]
This is because initially it was called "wp-notify," but then we realized that there was already a plugin with the same name and it was renamed "wp-notification." I won't hide the fact that I much prefer wp-notify, but the snag would be that after installing the plugin it would advise you to update it and when you do it replaces it [with this](https://wordpress.org/plugins/wp-notify/)

I think therefore we are forced to keep WP_NOTIFICATION unless we convince the author of wp-notify to give us the name

**Comment by johnhooks** [comment]
I agree notifications is a really long word. Though if we commit to PHP namespaces, it's not so bad. `WP/Notification` would only be at top of the file in uses statemenst, not part of every name.

**Comment by johnhooks** [comment]
~I support using namespaces and PSR-4 autoloading, and worry about how it will work inside core, when/if we get there.~

Should probably do it how WordPress core would.

Also I checked all of these have an old plugin parked on them:

- [wp-notice](https://wordpress.org/plugins/wp-notice/)
- [wp-notify](https://wordpress.org/plugins/wp-notify/)
- [wp-notification](https://wordpress.org/plugins/wp-notification/)
- [wp-notifications](https://wordpress.org/plugins/wp-notifications/) - this one says it's closed does that mean available?

**Comment by Sephsekla** [comment]
I'd like to use namespaces rather than prefixes where possible, it's very very stable PHP at this stage and I'd like to use this as an opportunity to get the ball rolling in core. We can always switch later.

**Comment by johnhooks** [comment]
- @Sephsekla just so we are clear, you support moving everything to namespaces and use the `WP/Notitifications`?
- If we stick with "Notifications" over "Notifiy", I think we need to do a big PR and remove "Notify", which would require renaming most the files in  `includes`. This is also related to the rest-api namespace #194.

**Comment by johnhooks** [comment]
There is also this, is the textdomain ` wp-feature-notifications`?

```php
/**
 * Plugin Name:       WP Feature Notifications
 * Plugin URI:        https://github.com/WordPress/wp-feature-notifications
 * Description:       A new (better) way to manage and deliver WordPress notifications to the relevant audience.
 * Version:           0.1.0
 * Requires at least: 5.0
 * Requires PHP:      7.4
 * Author:            WP Feature Notifications Contributors
 * Text Domain:       wp-feature-notifications
 * License:           GPL v2 or later
 * License URI:       http://www.gnu.org/licenses/gpl-2.0.txt
 *
 * @package wp-feature-notifications
 */
```

**Comment by johnhooks** [comment]
Fixed by PR #245.
